### PR TITLE
Added functionality to edit the Billpayer information for orders

### PR DESCRIPTION
**Pull request by molnarerwin**
### Description:
- Added functionality to edit the Billpayer information for orders.

### Screenshots:
image
image

### CODE REVIEW

**Review by fulopattila122 — APPROVED** [review]
Besides the BCs it's a good one :+1: 

I'll revert those BCs and merge the PR

File: src/resources/views/order/show/_update_billpayer_modal.blade.php [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+            {!! Form::close() !!}
+        </div>
+    </div>
+</div>
+
+@push('scripts')
+    <script>
+        document.addEventListener('alpine:init', () => {
+            Alpine.data('editBillpayerModal', () => ({
+                isOrganization: @json(old('is_organization') === null && $billpayer->is_organization) || {{ old('billpayer.is_organization') == '1' ? 'true' : 'false' }} || @json(!empty($billpayer->company_name)),
```
Comment (fulopattila122) [inline src/resources/views/order/show/_update_billpayer_modal.blade.php:208]: In what case do we have `is_organization`? Isn't it always `billpayer.is_organization`? :thinking:

File: src/Http/Requests/UpdateOrder.php [hunk 2]
```diff
@@ -42,4 +66,38 @@ public function authorize()
...
+    }
+
+    protected function prepareForValidation(): void
+    {
+        if ($this->has('billpayer') && !$this->isOrganization()) {
+            $this->merge([
+                'billpayer' => array_merge($this->input('billpayer', []), [
+                    'company_name' => null,
+                    'registration_nr' => null,
+                    'tax_nr' => null,
```
Comment (fulopattila122) [inline src/Http/Requests/UpdateOrder.php:98]: :+1: :+1: :+1:

File: src/resources/views/order/show/_update_billpayer_modal.blade.php (hunk 1 above)
Comment (fulopattila122) [inline src/resources/views/order/show/_update_billpayer_modal.blade.php:208]: I'm not 100% sure TBH, but I think this is a better condition
```suggestion
                isOrganization: @json(old('billpayer.is_organization') == '1' || !empty($billpayer->company_name)),
```

File: src/Http/Requests/UpdateOrder.php [hunk 3]
```diff
@@ -13,26 +13,50 @@
...
+            'billpayer.address.address' => 'required_with:billpayer|min:12|max:384',
+            'billpayer.address.postalcode' => 'nullable|min:4|max:12',
+            'billpayer.address.city' => 'nullable|min:2|max:255',
+            'billpayer.address.country_id' => ['required_with:billpayer', 'alpha:ascii', 'size:2', 'exists:countries,id'],
         ];
     }
 
     public function wantsToChangeOrderStatus(Order $order): bool
     {
+        if (!$this->has('status') {
```
Comment (fulopattila122) [inline src/Http/Requests/UpdateOrder.php:48]:
```suggestion
        if (!$this->has('status')) {
```

File: src/Contracts/Requests/UpdateOrder.php [hunk 4]
```diff
@@ -20,5 +20,7 @@ interface UpdateOrder extends BaseRequest
 {
     public function wantsToChangeOrderStatus(Order $order): bool;
 
-    public function getStatus(): string;
+    public function getStatus(): ?string;
+
+    public function wantsToUpdateBillpayerData(): bool;
```
[the remark this answers is not in the document]
Comment (molnarerwin) [reply src/Contracts/Requests/UpdateOrder.php]: Thank you for pointing this out! I’ll make sure to be more careful about these kinds of changes in the future.

**Pull request merged**